sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/LeaseClient.java

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.specialized;
+
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.VoidResponse;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.ContainerClient;
+import com.azure.storage.blob.models.ModifiedAccessConditions;
+import com.azure.storage.common.Utility;
+
+import java.net.URL;
+import java.time.Duration;
+
+/**
+ *
+ */
+public final class LeaseClient {
+    private final LeaseAsyncClient client;
+
+    LeaseClient(LeaseAsyncClient client) {
+        this.client = client;
+    }
+
+    /**
+     * Gets the {@link URL} of the lease client. If the lease client is using a {@link BlobClient} it will be the URL of
+     * the blob, otherwise the URL will be the {@link ContainerClient} URL.
+     *
+     * @return URL of the lease client.
+     */
+    public URL getLeaseClientUrl() {
```

Comment:
Apply same comments from the async client here also.

Reply:
`getLeaseUrl`